src/mrseq/scripts/radial_flash.py

```diff
@@ -0,0 +1,405 @@
+
+            # add delay in case TR > min_TR
+            if tr_delay > 0:
+                seq.add_block(pp.make_delay(tr_delay))
+
+            if mrd_header_file and spoke_ >= 0:
+                # add acquisitions to metadata
+                k_radial_line = np.linspace(
+                    -n_readout_with_oversampling // 2,
+                    (n_readout_with_oversampling // 2) - 1,
+                    n_readout_with_oversampling,
+                )
+                radial_trajectory = np.zeros((n_readout_with_oversampling, 2), dtype=np.float32)
+
+                radial_trajectory[:, 0] = k_radial_line * np.cos(rotation_angle_rad)
+                radial_trajectory[:, 1] = k_radial_line * np.sin(rotation_angle_rad)
+
+                acq = ismrmrd.Acquisition()
+                acq.resize(trajectory_dimensions=2, number_of_samples=adc.num_samples)
+                acq.traj[:] = radial_trajectory
+                prot.append_acquisition(acq)
+
+    # close ISMRMRD file
+    if mrd_header_file:
+        prot.close()
+
+    return seq, min_te, min_tr
+
+
+def main(
+    system: pp.Opts | None = None,
+    te: float | None = None,
+    tr: float | None = None,
+    fov_xy: float = 128e-3,
+    n_readout: int = 128,
+    n_spokes: int = 128,
+    slice_thickness: float = 8e-3,
+    n_slices: int = 1,
+    receiver_bandwidth_per_pixel: float = 800,  # Hz/pixel
+    show_plots: bool = True,
+    test_report: bool = True,
+    timing_check: bool = True,
+) -> pp.Sequence:
+    """Generate a radial FLASH sequence.
+
+    Parameters
+    ----------
+    system
+        PyPulseq system limits object.
+    te
+        Desired echo time (TE) (in seconds). Minimum echo time is used if set to None.
+    tr
+        Desired repetition time (TR) (in seconds). Minimum repetition time is used if set to None.
+    fov_xy
+        Field of view in x and y direction (in meters).
+    n_readout
+        Number of frequency encoding steps.
+    n_spokes
+        Number of radial lines.
+    slice_thickness
+        Slice thickness of the 2D slice (in meters).
+    n_slices
+        Number of slices.
+    receiver_bandwidth_per_pixel
+        Desired receiver bandwidth per pixel (in Hz/pixel). This is used to calculate the readout duration.
+    show_plots
+        Toggles sequence plot.
+    test_report
+        Toggles advanced test report.
+    timing_check
+        Toggles timing check of the sequence.
+    """
+    if system is None:
+        system = sys_defaults
+
+    # define settings of rf excitation pulse
+    rf_duration = 1.28e-3  # duration of the rf excitation pulse [s]
+    rf_flip_angle = 12  # flip angle of rf excitation pulse [°]
+    rf_bwt = 4  # bandwidth-time product of rf excitation pulse [Hz*s]
+    rf_apodization = 0.5  # apodization factor of rf excitation pulse
+    readout_oversampling = 2  # readout oversampling factor, commonly 2. This reduces aliasing artifacts.
+    spoke_angle = np.pi / 180 * (180 * 0.618034)
```

Comment:
do we maybe want to define the different golden angles for full spoke and half spoke / spiral etc as reusable constants in mrseq?